### PR TITLE
Remove usages of WindowManager's getDefaultDisplay

### PR DESCRIPTION
**Pull request by jesswrd**
Removed deprecated usages of WindowManager's `getDefaultDisplay()` and replaced them with DisplayManager's `getDisplay()`.

Note: Decided to keep a usage of `getDefaultDisplay()`, which can be found in FlutterView.java because it is expected to be deleted as a part of V1 embedding removal. No changes were made to that file. 

path to FlutterView.java: shell/platform/android/io/flutter/embedding/android/FlutterView.java

Fixes [#99421](https://github.com/flutter/flutter/issues/99421)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by yaakovschectman — APPROVED** [review]
LGTM

**Comment by reidbaker** [comment]
before landing remember to update the pre launch checklist in the pull request body.

**Comment by reidbaker** [comment]
Should this pr be held until https://github.com/flutter/flutter/issues/154746 is resolved?

File: shell/platform/android/io/flutter/Build.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 public class Build {
   /** For use in place of the Android Build.VERSION_CODES class. */
   public static class API_LEVELS {
+    public static final int FLUTTER_MIN = 21;
```
Comment (reidbaker) [inline shell/platform/android/io/flutter/Build.java:11]: We should add some documentation for why this value exists and add pointers to it from the other places we change the minimum value. Also I think we should make this value visible for testing.  I think in production/shipped code we want to set an actual bound and if minimum value gets higher than a split review that code and remove one branch. But for tests this lets us say "this method should support all api versions that flutter supports". 

https://github.com/flutter/flutter/blob/b5749a7d65be14d655a3de4fe9d82cc758c3b85e/docs/platforms/android/New-Android-version.md?plain=1 is a good starting point for finding where we might need to link to and from this file. 

// The minimum android api that flutter supports. Intended for use in tests that don't actually care about the api level but want to set a minSdk bound that does not need to regularly be updated when the minimum version changes.

**Review by reidbaker — APPROVED** [review]
Approved % api level documentation and visibility change.

File: shell/platform/android/io/flutter/Build.java (hunk 1 above)
Comment (jesswrd) [reply shell/platform/android/io/flutter/Build.java]: Doc is located in the flutter/flutter repo. Will make a different PR to update the docs.

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java [hunk 2]
```diff
@@ -597,26 +597,36 @@ protected void setWindowInfoListenerDisplayFeatures(WindowLayoutInfo layoutInfo)
...
     BOTH
   }
 
-  private ZeroSides calculateShouldZeroSides() {
+  /**
+   * This method can be run on APIs 30 and above but its intended use is for 30 and below.
+   *
+   * @return some ZeroSides enum
+   */
+  @SuppressWarnings("deprecated")
```
Comment (gmackall) [inline shell/platform/android/io/flutter/embedding/android/FlutterView.java:613]: fyi I think this needs to be
```suggestion
  @SuppressWarnings("deprecation")
```

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java [hunk 3]
```diff
@@ -597,26 +597,36 @@ protected void setWindowInfoListenerDisplayFeatures(WindowLayoutInfo layoutInfo)
...
     int orientation = context.getResources().getConfiguration().orientation;
-    int rotation =
-        ((WindowManager) context.getSystemService(Context.WINDOW_SERVICE))
-            .getDefaultDisplay()
-            .getRotation();
 
     if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+      int rotation =
+          ((DisplayManager) context.getSystemService(Context.DISPLAY_SERVICE))
+              .getDisplay(0)
```
Comment (gmackall) [inline shell/platform/android/io/flutter/embedding/android/FlutterView.java:627]: nit: intent might be clearer as
```suggestion
              .getDisplay(Display.DEFAULT_DISPLAY)
```
https://developer.android.com/reference/android/view/Display#constants_1

**Review by gmackall — APPROVED** [review]

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java (hunk 2 above)
Comment (gmackall) [reply shell/platform/android/io/flutter/embedding/android/FlutterView.java:613]: at least I assume that's why the lint check is failing

**Comment by gmackall** [comment]
nit: remove [wip] from title before merging

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java (hunk 2 above)
Comment (jesswrd) [reply shell/platform/android/io/flutter/embedding/android/FlutterView.java]: done. Changed baseline.xml file to suppress linter

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java (hunk 3 above)
Comment (jesswrd) [reply shell/platform/android/io/flutter/embedding/android/FlutterView.java]: done.

**Comment by jesswrd** [comment]
> nit: remove [wip] from title before merging

done.

**Pull request merged**